### PR TITLE
-V option added

### PR DESCRIPTION
**Pull request by div-sahu55**

### CODE REVIEW

**Comment by anjanik012** [comment]
This seems fine but there is a much better way to do this by using CMake. Try it yourself. If you couldn't do it then I'll tell you how to

**Comment by div-sahu55** [comment]
> This seems fine but there is a much better way to do this by using CMake. Try it yourself. If you couldn't do it then I'll tell you how to
I tried using the project() variable in cmake to set the version but i cant figure out how to access that version from main function.

**Comment by RedocamaI** [comment]
> > This seems fine but there is a much better way to do this by using CMake. Try it yourself. If you couldn't do it then I'll tell you how to
> > I tried using the project() variable in cmake to set the version but i cant figure out how to access that version from main function.

I think this might help you: https://cmake.org/cmake/help/latest/guide/tutorial/index.html

**Comment by anjanik012** [comment]
@div-sahu55 You should take a look at this. https://cmake.org/cmake/help/latest/guide/tutorial/index.html#adding-a-version-number-and-configured-header-file

**Comment by div-sahu55** [comment]
> @div-sahu55 You should take a look at this. https://cmake.org/cmake/help/latest/guide/tutorial/index.html#adding-a-version-number-and-configured-header-file

Yes sir, just updated the PR.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,7 @@
 cmake_minimum_required(VERSION 3.13)
```
Comment (anjanik012) [inline CMakeLists.txt:1]: This PR seems to work fine. However, before merging we need to clean this up a little.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
-
-project(passgen LANGUAGES CXX)
-
+project(passgen VERSION 1.0.0 LANGUAGES CXX)
```
Comment (anjanik012) [inline CMakeLists.txt]: A more Ideal way is to set CMake variables to store MAJOR, MINOR and REVISION numbers. Having version in top level CMakeLists.txt is ideal.
Example:-

```
set(passgen_VERSION MAJOR 1)
set(passgen_VERSION_MINOR 0)
set(passgen_VERSION_REVISION 0)

```
No need for VERSION in `project` command here

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,9 +1,7 @@
...
+configure_file(Config.h.in Config.h)
```
Comment (anjanik012) [inline CMakeLists.txt]: version.h is a better filename

File: Config.h.in [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#define passgen_VERSION_MAJOR 1
```
Comment (anjanik012) [inline Config.h.in]: This file should be in src/

Comment (anjanik012) [inline Config.h.in]: Replace Hardcoded values here with the variables defined in the top-level CMakeLists.txt as

```
#define passgen_VERSION_MAJOR @passgen_VERSION_MAJOR@
#define passgen_VERSION_MINOR @passgen_VERSION_MINOR@
#define passgen_VERSION_REVISION @passgen_VERSION_REVISION@
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,9 +1,7 @@
...
 set(CMAKE_CXX_STANDARD 17)
 
 add_executable(passgen ${PROJECT_SOURCE_DIR}/src/main.cpp)
-
-
+target_include_directories(passgen PUBLIC "${PROJECT_BINARY_DIR}")
```
Comment (anjanik012) [inline CMakeLists.txt]: Path will change to ${PROJECT_BINARY_DIR}/src

File: src/main.cpp [hunk 6]
```diff
@@ -1,6 +1,17 @@
 #include <iostream>
+#include "Config.h"
 
-int main(int argc, char *argv[]){
-    std::cout << "Hello,World!" << std::endl;
+int main(int argc, char *argv[])
+{
```
Comment (anjanik012) [inline src/main.cpp]: Please don't use opening braces in a new line

File: src/main.cpp [hunk 7]
```diff
@@ -1,6 +1,17 @@
...
+
+    for (int i = 0; i < argc; ++i)
+    {
+        if (std::string(argv[i]) == "-V")
+        {
+            std::cout << "Current version: "
```
Comment (anjanik012) [inline src/main.cpp]: Replace Current Version with "passgen version: "

File: src/main.cpp [hunk 8]
```diff
@@ -1,6 +1,17 @@
...
+                      << passgen_VERSION_MAJOR << "." << passgen_VERSION_MINOR
```
Comment (anjanik012) [inline src/main.cpp]: Also print the REVISION number

File: CMakeLists.txt (hunk 3 above)
Comment (anjanik012) [inline CMakeLists.txt]: After moving this file to src/ don't forget to change the path here as well.

File: src/main.cpp [hunk 9]
```diff
@@ -1,16 +1,12 @@
 #include <iostream>
-#include "Config.h"
-
-int main(int argc, char *argv[])
-{
+#include "version.h"
 
+int main(int argc, char *argv[]){
     for (int i = 0; i < argc; ++i)
     {
```
Comment (anjanik012) [inline src/main.cpp]: opening brace on a new line

File: src/main.cpp [hunk 10]
```diff
@@ -1,16 +1,12 @@
...
         if (std::string(argv[i]) == "-V")
         {
```
Comment (anjanik012) [inline src/main.cpp]: opening brace on a new line

File: src/main.cpp (hunk 9 above)
Comment (anjanik012) [inline src/main.cpp]: Opening brace in new line

**Comment by anjanik012** [comment]
It is ready to merge however  you need to rebase your PR and resolve merge conflict

**Comment by anjanik012** [comment]
Also while resolving merge conflict, use `strcmp` function as used by Suraj to compare string instead of making a string object.

**Pull request merged**